### PR TITLE
added an optional plugin which control sampling by specifying a "percentage" instead of "agent.sample_n_per_3_secs"

### LINKED ISSUE #5997: agent side percentage sampling config option
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [X] Question or discussion
- [ ] Bug
- [ ] Requirement
- [ ] Feature or performance improvement

___
### Question
- What do you want to know?
For sampling, it seems currently (v8.2.0/v8.3.0) there are primarily 2 ways to control trace sampling. One is in server side using `sampleRate` but the bandwidth is still wasted and may not be very flexible if has many multiple sources of agent. The other is via agent side `agent.sample_n_per_3_secs` option but the use case is kind of counter-intuitive. In many scenarios one may want a percentage sampling rate option in agent side. What's the reason of not providing that but using a "traces in 3 seconds" mechanism instead? This looks a bit odd and from looking at the `SamplingService` code a percentage way would not be difficult to implement.   
___
### Bug
- Which version of SkyWalking, OS and JRE?
v8.2.0
openjdk version "1.8.0_265"

___
### Requirement or improvement
- provide an agent config similar to `agent.sample_n_per_3_secs` but in percentage (like 10 in every 10000). We can try to implement this and open a PR if that's OK.

### PR DESCRIPTION
**Pull request by Frefreak**
This trivial plugin provides a `plugin.sampling.sample_rate` config to control sampling rate (n out of 10000). It overrides default `SamplingService`. Our team is using it to work around what had been discussed in #5997 and thought this might be useful to some others.

### Add an agent plugin to support <framework name>
- [x] Add a test case for the new plugin, refer to [the doc](https://github.com/apache/skywalking/blob/master/docs/en/guides/Plugin-test.md)
- [ ] Add a component id in [the component-libraries.yml](https://github.com/apache/skywalking/blob/master/oap-server/server-bootstrap/src/main/resources/component-libraries.yml)
- [ ] Add a logo in [the UI repo](https://github.com/apache/skywalking-rocketbot-ui/tree/master/src/views/components/topology/assets)
     

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #5997.
- [ ] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
I am confused, during the discussion, I said https://github.com/apache/skywalking/issues/5997#issuecomment-743729015. I can't see the question had been answered and provided reasonable explanation.

**Comment by Frefreak** [comment]
For example, imagine in a scenario where there is an operation's QPS changes during the day, say 10000 in the evening and 100 in the morning. If I want a portion of all the samples to be sampled, a fixed `sample_n_per_3_secs` would not work very well because it's kind of hard to determine what value should be used for it. If I set it relative to QPS = 100, too less samples would be sampled when QPS = 10000. If I set it relative to QPS = 10000, all the traces in other time might be sampled. Besides I need know the current QPS in order to set it right. In this case if one can just specify a number indicating the percentage rate, it would be way easier to achieve this.

Also we do understand that the sample rate would not be accurate, that's expected. This plugin doesn't intend to be accurate either.
In summary this plugin just provides another way to specify how often one would like traces to be sampled, please consider its use cases. Not all users need this of course, but maybe some users would consider this useful.

**Comment by wu-sheng** [comment]
> If I set it relative to QPS = 100, too less samples would be sampled when QPS = 10000. If I set it relative to QPS = 10000, all the traces in other time might be sampled.

Personally, I can't see the real value of this. Because of the inaccurate issue you have mentioned, the agent-side sampling only makes sense when you are facing high-traffic-load. Which percent-sampling mechanism doesn't really help. When the traffic goes to over 10+ times than it usually is, sampling rate keeps unchanged, and much more data has been sampled.

**Comment by pierre94** [comment]
> Personally, I can't see the real value of this. Because of the inaccurate issue you have mentioned, the agent-side sampling only makes sense when you are facing high-traffic-load. Which percent-sampling mechanism doesn't really help. When the traffic goes to over 10+ times than it usually is, sampling rate keeps unchanged, and much more data has been sampled.

I think sampling by specifying a "percentage" can be a optional configuration for users. 

Jaeger provides a similar configuration[`Probabilistic`](https://www.jaegertracing.io/docs/1.21/sampling/#client-sampling-configuration)

**Comment by wu-sheng** [comment]
`Jaeger` option belongs to them. We are asking for use case, rather than referencing. SkyWalking should always focus on the use case, what benefits users could have, rather than what configurations other projects have.

**Comment by wu-sheng** [comment]
We are moving fast on Agent CDS(Config Discovery Service) https://github.com/apache/skywalking/issues/6114.
With that, it also reduces the requirement of using percentage mechanism to do sampling. Actually, you could do automatical sampling easily through that.

**Comment by Frefreak** [comment]
Yes, dynamic configuration will do. We're recently also experimenting with that but official support would be optimal. Looking forward to that. Thanks!
